kernel/fatal.c

```diff
@@ -89,8 +89,7 @@ void z_fatal_error(unsigned int reason, const struct arch_esf *esf)
 	 * appropriate.
 	 */
 	unsigned int key = arch_irq_lock();
-	struct k_thread *thread = IS_ENABLED(CONFIG_MULTITHREADING) ?
-			_current : NULL;
+	struct k_thread *thread = _current;
```

Comment:
There is no need to change this since you are adding a guard to LOG_ERR() below.

Reply:
done